aesd-char-driver/main.c

```diff
@@ -163,11 +165,93 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
 	mutex_unlock(&device->char_device_lock);	
 	return retval;
 }
+
+loff_t aesd_llseek(struct file *filp, loff_t offset, int whence)
+{
+	loff_t ret_val;
+	struct aesd_dev *dev = filp->private_data;
+	loff_t size = dev->aesd_buffer.filled_buffer_size;
+	//include lock
+	if(mutex_lock_interruptible(&dev->char_device_lock))
+		return -ERESTARTSYS; 
+	ret_val = fixed_size_llseek(filp, offset, whence, size);
```

Comment:
@Ganeshkm10 what is `char_device_lock` locking?  Is it necessary to hold this lock here?